testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/deployment/NonExistingJsfImplDeploymentTestCase.java

```diff
@@ -0,0 +1,97 @@
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.test.shared.ServerReload;
+import org.jboss.as.test.shared.SnapshotRestoreSetupTask;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup(NonExistingJsfImplDeploymentTestCase.SetupTask.class)
+public class NonExistingJsfImplDeploymentTestCase {
+
+    @ContainerResource
+    private ManagementClient managementClient;
+
+    @ArquillianResource
+    private Deployer deployer;
+
+    @Deployment(testable = false, managed = false, name = "test_jar")
+    public static Archive<?> deploy() {
+        JavaArchive archive = ShrinkWrap.create(JavaArchive.class);
+        archive.addClass(NonExistingJsfImplDeploymentTestCase.class);
+        return archive;
+    }
+
+    private static final PathAddress RA_ADDRESS = PathAddress.pathAddress(ModelDescriptionConstants.SUBSYSTEM, "jsf");
+
+    @Test
+    public void testDeploymentDoesntFailBecauseOfNPE() throws Exception {
+        try {
+            deployer.deploy("test_jar");
+        } catch (Exception e) {
+            if (e instanceof DeploymentException) {
+                // awkward way to check if the failure is caused by NPE, but e.getCause() doesn't work
+                StringWriter out = new StringWriter();
+                e.printStackTrace(new PrintWriter(out));
+                String stackTrace = out.getBuffer().toString();
+
+                if (stackTrace.contains("Caused by: java.lang.NullPointerException")) {
+                    throw e;
+                }
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    static class SetupTask extends SnapshotRestoreSetupTask {
+        @Override
+        protected void doSetup(ManagementClient client, String containerId) throws Exception {
+            ModelControllerClient mcc = client.getControllerClient();
+            ModelNode addRaOperation = Operations.createWriteAttributeOperation(RA_ADDRESS.toModelNode(), "default-jsf-impl-slot", "idontexist");
```

Comment:
Minor nit but `addRaOperation` should be reworded to something like `addJSFOperation`.